recipes/avahi/all/conanfile.py

```diff
@@ -26,75 +30,68 @@ class AvahiConan(ConanFile):
         "fPIC": True
     }
 
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("glib/2.68.3")
-        self.requires("expat/2.4.1")
+        self.requires("glib/2.74.0")
+        self.requires("expat/2.4.9")
         self.requires("libdaemon/0.14")
-        self.requires("dbus/1.12.20")
+        self.requires("dbus/1.15.0")
         self.requires("gdbm/1.19")
         self.requires("libevent/2.1.12")
 
     def validate(self):
-        if self.settings.os != "Linux" or tools.cross_building(self):
+        if self.info.settings.os != "Linux":
             raise ConanInvalidConfiguration("Only Linux is supported for this package.")
 
     def configure(self):
-        del self.settings.compiler.cppstd
-        del self.settings.compiler.libcxx
+        try:
+            del self.settings.compiler.cppstd
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    @property
-    def _configure_args(self):
-        yes_no = lambda v: "yes" if v else "no"
-        return [
-            "--enable-shared={}".format(yes_no(self.options.shared)),
-            "--enable-static={}".format(yes_no(not self.options.shared)),
-            "--disable-gtk3",
-            "--disable-mono",
-            "--disable-python",
-            "--disable-qt5",
-            "--disable-monodoc",
-            "--enable-compat-libdns_sd",
-            "--with-systemdsystemunitdir={}/lib/systemd/system".format(self.package_folder),
-        ]
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self)
-        self._autotools.configure(configure_dir=self._source_subfolder, args=self._configure_args)
-        return self._autotools
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.append("--enable-compat-libdns_sd")
+        tc.configure_args.append("--disable-gtk3")
+        tc.configure_args.append("--disable-mono")
+        tc.configure_args.append("--disable-monodoc")
+        tc.configure_args.append("--disable-python")
+        tc.configure_args.append("--disable-qt5")
+        tc.configure_args.append("--with-systemdsystemunitdir=/lib/systemd/system")
```

Comment:
Noob question, why is it not relative to package folder? Should this be an option?

Reply:
It's how the package uses `DESTDIR` when installing. It ends up using appending this path to `DESTDIR` which is the package directory already. By incorporating the `self.package_folder`, the package directory path is duplicated and this ends up in a very strange place in the package folder. I believe the way the new `AutotoolsToolchain` handles `DESTDIR` is why this change was necessary. As for making it an option, I'm not sure that's necessary. Putting it the standard directory path seems to make the most sense right now since it keeps the previous behavior of the package. Maybe is someone specifically requests this be an option in the future, it can be added then.

Reply:
Thanks for the amazing answer!